File: src/pokedata/Pokemon.cs
using System.Text.Json.Serialization;
using Pokedex.pokedata;

namespace Pokedex;

public class Pokemon
{
    [JsonPropertyName("abilities")]
    public List<Abilities> Abilities { get; set; }
    [JsonPropertyName("base_experience")]
    public ushort BaseExperience { get; set; }
    [JsonPropertyName("forms")]
    public List<Form> Forms { get; set; }
    [JsonPropertyName("game_indices")]
    public List<GameIndex>? GameIndices { get; set; }
    [JsonPropertyName("height")]
    public ushort Height { get; set; }
    [JsonPropertyName("held_items")]
    public List<HeldItem>? HeldItems { get; set; }
    [JsonPropertyName("id")]
    public int Id { get; set; }
    [JsonPropertyName("is_default")]
    public bool IsDefault { get; set; }
    [JsonPropertyName("encounters")]
    public string? Encounters { get; set; }
    [JsonPropertyName("moves")]
    public List<Moves> Moves { get; set; }
    [JsonPropertyName("name")]
    public string Name { get; set; }
    [JsonPropertyName("order")]
    public ushort Order { get; set; }
    [JsonPropertyName("past_types")]
    public List<object>? PastTypes { get; set; }
    [JsonPropertyName("species")]
    public Species Species { get; set; }
    [JsonPropertyName("sprites")]
    public Sprites Sprites { get; set; }
    [JsonPropertyName("stats")]
    public List<Stat> Stats { get; set; }
    [JsonPropertyName("weight")]
    public ushort Weight { get; set; }
}

File: PokedexFs/ClientApp/src/pokedataft/Pokemon.ts
import { List } from "../data/list"
import { Abilities } from "./Abillities";
import { Form } from "./IForm";
import { GameIndex } from "./IGameIndex";
import { Moves } from "./Moves";
import { HeldItem } from "./IHeldItem";
import { Species} from "./ISpecies";
import { Sprites } from "./ISprites";
import { Stat } from "./IStat";

export class Pokemon {
    
    static Abilities: List<Abilities>;
    static BaseExperience: number;
    static Forms: List<Form>;
    static GameIndices: List<GameIndex> | null;
    static Height: number;
    static HeldItems: List<HeldItem> | null;
    static Id: number ;
    static IsDefault: boolean;
    static Encounters: string | null;
    static Moves: List<Moves>;
    static Name: string;
    static Order: number;
    static PastTypes: List<object> | null;
    static Species: Species;
    static Sprites: Sprites;
    static Stats: List<Stat>;
    static Weight: number;
}